BlueShift-iOS-SDK/BlueShift.m

```diff
@@ -28,7 +38,13 @@ + (instancetype) sharedInstance {
 
 #pragma mark SDK initialisation
 + (void) initWithConfiguration:(BlueShiftConfig *)config {
-    [[BlueShift sharedInstance] setupWithConfiguration:config];
+    if([NSThread isMainThread] == YES) {
+        [[BlueShift sharedInstance] setupWithConfiguration:config];
```

Comment:
why do we need to push this to run in the main thread?

Reply:
Yes, there is some code that requires to be executed on the main thread.